src/backend/distributed/commands/utility_hook.c

```diff
@@ -612,6 +658,180 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 		 */
 		CitusHasBeenLoaded();
 	}
+
+	if (list_length(ddlJobs) != 0)
+	{
+		return true;
+	}
+
+	if (IsDropSchemaOrTable(parsetree))
+	{
+		/*
+		 * TODO: We could specifically check if it is a:
+		 * - DROP TABLE statement that drops a reference or a citus local table
+		 *   that has a foreign key relationship with a postgres table, or a
+		 * - DROP SCHEMA statement that again drops such a citus table
+		 * but this is more easy and the other approach is much more prune to
+		 * deadlocks.
+		 */
```

Comment:
We also need to handle `DROP OWNED BY ..` similar to `IsDropSchemaOrTable `

Reply:
now we don't need to handle such commands separately but I need to add tests as mentioned in https://github.com/citusdata/citus/pull/4538#discussion_r561928566, so resolving this one